### PR TITLE
optimize leader signature collection strategy; Clean up logs

### PR DESCRIPTION
**Pull request by rlan35**
Leader sig collection strategy:

**before**: once 2/3 voting power/sig collected, wait for another 2s to collect more sigs; or if 90% reached, just commit.

**now**: 
    1. always wait for the full block time (8s) and collect as much sigs as possible. New block won't be proposed before full block time.
    2. once 2/3 voting power/sig collected, wait for another 2s or more if full block time not reached.
    3. If 100% sig collected, commit the block immediately
    4. Both 2 and 3 works under constraint of 1

This change will help improve the chance of slow node getting their signatures collected in block

### CODE REVIEW

File: consensus/consensus_v2.go [hunk 1]
```diff
@@ -473,12 +470,19 @@ func (consensus *Consensus) Start(
 
 			case viewID := <-consensus.commitFinishChan:
 				consensus.getLogger().Debug().Msg("[ConsensusMainLoop] commitFinishChan")
+
 				// Only Leader execute this condition
 				func() {
+					if time.Now().Before(blockTime) {
```
Comment (fxfactorial) [inline consensus/consensus_v2.go:476]:
```
if n := time.Now(); n.Before(blockTime) {

time.Sleep(blockTime.Sub(n)) 

}
```

unless you are intentionallyy making a fresh `time.Now()`?

**Comment by fxfactorial** [comment]
This PR actually has quite some nuanced logic in consensus and thresholding. Can we test on STN before merge to master?

File: consensus/consensus_v2.go (hunk 1 above)
Comment (rlan35) [reply consensus/consensus_v2.go]: good point. done.

**Comment by rlan35** [comment]
> This PR actually has quite some nuanced logic in consensus and thresholding. Can we test on STN before merge to master?

I've already tested locally, and consensus works fine. Won't hurt to merge and test with STN.

**Comment by fxfactorial** [comment]
> > This PR actually has quite some nuanced logic in consensus and thresholding. Can we test on STN before merge to master?
> 
> I've already tested locally, and consensus works fine. Won't hurt to merge and test with STN.

fair - I agree

File: consensus/quorum/one-node-one-vote.go [hunk 2]
```diff
@@ -52,9 +52,9 @@ func (v *uniformVoteWeight) QuorumThreshold() numeric.Dec {
 	return numeric.NewDec(v.TwoThirdsSignersCount())
 }
 
-// RewardThreshold ..
-func (v *uniformVoteWeight) IsRewardThresholdAchieved() bool {
-	return v.SignersCount(Commit) >= (v.ParticipantsCount() * 9 / 10)
+// IsAllSigsCollected ..
+func (v *uniformVoteWeight) IsAllSigsCollected() bool {
+	return v.SignersCount(Commit) == v.ParticipantsCount()
```
Comment (fxfactorial) [inline consensus/quorum/one-node-one-vote.go:57]: but this will make it easier for consensus to break and p2p hasn't been worked on yet, why bump it up prematurely?

**Comment by rlan35** [comment]
> in principle approve but the bump up from 90% to 100% i think will cause problems

The 90% or 100% is a second tier criteria. The first tier is 2/3 and wait for 2s for any more sigs.

100% won't matter when first tier works. So should be a pure win situation.

**Comment by LeoHChen** [comment]
Sorry, I'm late for reviewing this PR. After reviewing the PR, I think this is the right direction to collection signatures.  If you don't mind, I still have a few questions even though the PR is merged.
1. do we always have to wait for the full block time before proposing the next block? what if all external nodes have signed the block already, then we don't need to wait. Noted, our own nodes have 2s commit delay already.
1. can we get @gupadhyaya review and approve this PR?
1. What would be a good test plan for this consensus update and block period update in STN? any corner case we shall cover?  we need to properly test it. @john-harmony , @SebastianJ

**Comment by rlan35** [comment]
> Sorry, I'm late for reviewing this PR. After reviewing the PR, I think this is the right direction to collection signatures. If you don't mind, I still have a few questions even though the PR is merged.
> 
> 1. do we always have to wait for the full block time before proposing the next block? what if all external nodes have signed the block already, then we don't need to wait. Noted, our own nodes have 2s commit delay already.
> 2. can we get @gupadhyaya review and approve this PR?
> 3. What would be a good test plan for this consensus update and block period update in STN? any corner case we shall cover?  we need to properly test it. @john-harmony , @SebastianJ

1. Yes, we still always wait for 8 seconds block time. It's same as before. So no change on this logic.
2. Yes, both Leo and Ganesha can take a extra look
3. This thing just needs to live through more test time with very test situation we do currently. It's hard to craft specific test case for it as it's distributed system thing. As long as there is no regression on block time or consensus stuck in next STN deploy, it's good to go.
